packages/flutter_tools/lib/src/runner/target_devices.dart

```diff
@@ -490,15 +510,40 @@ class TargetDevicesWithExtendedWirelessDeviceDiscovery extends TargetDevices {
         includeDevicesUnsupportedByProject: includeDevicesUnsupportedByProject,
         includeDisconnected: true,
       );
+
       if (devices.length == 1) {
         Device? matchedDevice = devices.first;
+
+        if (matchedDevice is IOSDevice && !matchedDevice.devModeEnabled) {
+          return _handleDisabledIosDevice(matchedDevice);
+        }
+
         if (!matchedDevice.isConnected && matchedDevice is IOSDevice) {
           matchedDevice = await _waitForIOSDeviceToConnect(matchedDevice);
         }
+
         if (matchedDevice != null && matchedDevice.isConnected) {
           return <Device>[matchedDevice];
         }
       }
+
+      if (devices.length > 1) {
+        final List<Device> disabledDevices = [];
+        for (final Device device in devices) {
+          if (device is IOSDevice && !device.devModeEnabled) {
+            disabledDevices.add(device);
+            if (disabledDevices.length == devices.length) {
+              // all matching devices have dev mode disabled, so print other
+              // available devices
+              return _handleDisabledIosDevice(device);
+            }
+            // otherwise, just show the error and move on
```

Comment:
i'm a bit confused - this says `otherwise` but the code below is not in an `else` block.

Reply:
this is cuz the if block above it returns, so it's basically an else block